os/arch/arm/src/armv7-a/arm_cpuhotplug.c

```diff
@@ -87,10 +87,24 @@ int arm_hotplug_handler(int irq, void *context, void *arg)
 
 	/* Set secondary core to hotplug state */
 	int cpu = this_cpu();
-	up_set_cpu_state(cpu, CPU_HOTPLUG);
-	/* Save the tcb state */
+	
 	struct tcb_s *rtcb = this_task();
-	arm_savestate(rtcb->xcp.regs);
+	if (up_get_cpu_state(cpu) == CPU_HALTED) {
+		/* 
+			clear GIC APR0 as when this gets set during restore by ATF boot flow,
+			When this register is set, it cause GIC to not serve other requests as it thinks there is an
+			active ISR being served currently.
+			
+			On powerup, this register is restored by ATF from previous saved state. 
+			If hard reset, then state is re-initialized, otherwise it will save the state when psci_cpu_off
+		*/
+		putreg32(0x00000000, GIC_ICCNSAPR1);	/* clear SGI1 nonsecure APR0 */
+		putreg32(0x00000000, GIC_ICCAPR1);	/* clear SGI1 secure APR0 */
```

Comment:
Hello henry 
shoud not reqiured clear SGI ARP when hotplug?

Reply:
the reason why there is no clear register in hotplug is because all the registers are cleared when entering and exiting PG?

Reply:
can we apply clear register in hotplug state during PG also?
because i think, 1. case also we should have clear register, but in the case, entering/exiting PG is covering that.

and, i think we have two option.

1) Clear register in any CPU1 off situation
2) move pcsi cpu off to cpu1 idle loop.

Could you please let me know your opinion?